app/src/main/java/com/google/samples/apps/sunflower/GardenFragment.kt

```diff
@@ -92,7 +89,24 @@ class GardenFragment : Fragment() {
     }
 
     private fun onSelectionChanged(selectedCount: Int) {
-        // TODO: Enable/disable ActionMode when items are selected.
+        if (selectedCount == 0) {
+            actionMode?.let {
+                it.finish()
+                actionMode = null
+            }
+        } else {
+            (requireActivity() as? AppCompatActivity)?.let { activity ->
```

Comment:
I don't think that `requireActivity()` returns `nullable`.
(Because of `nullable` by `getActivity()`, Google provides `requireActivity()` which doesn't return `null`, as I know.)

Reply:
You're right, it's not `nullable`, but it returns a `FragmentActivity`, and we need an `AppCompatActivity`. The `as?` does the, try to cast this to an `AppCompatActivity`. If it *can't* (I don't know why it wouldn't, really), then, rather than throwing, make the value `null`. :)

Reply:
:point_up: that's a great explanation.  Add it as a comment?